common/post-processing-worker-filter.h

```diff
@@ -22,17 +22,31 @@ class post_processing_worker_filter : public post_processing_filter
     }
     ~post_processing_worker_filter()
     {
-        _alive = false;
-        if( _worker.joinable() )
-            _worker.join();
+        if (_alive)
+        {
+            _alive = false;
+            if (_worker.joinable())
+            {
+                if (std::this_thread::get_id() != _worker.get_id())
```

Comment:
`// joinable() only checks that the thread's alive -- but we don't want to join if we're here from within the worker thread itself!`